code/parachain/frame/oracle/src/lib.rs

```diff
@@ -836,26 +836,38 @@ pub mod pallet {
 			}
 			if let Some(mut reward_tracker) = Self::get_reward_tracker_if_enabled() {
 				// divide the per asset reward(by weight) by the number of oracles
-				let reward_amount_per_oracle: T::Balance = safe_multiply_by_rational(
+				let reward_amount_per_asset: T::Balance = safe_multiply_by_rational(
 					reward_tracker.current_block_reward.unique_saturated_into(),
 					asset_info.reward_weight.unique_saturated_into(),
 					reward_tracker.total_reward_weight.unique_saturated_into(),
 				)?
-				.checked_div(rewarded_oracles.len() as u128)
-				.ok_or(ArithmeticError::DivisionByZero)?
 				.into();
-				if !reward_amount_per_oracle.is_zero() {
-					for accounts in rewarded_oracles {
-						Self::transfer_reward(
-							accounts.0,
-							accounts.1,
-							asset_id,
-							reward_amount_per_oracle,
-						)?;
-						// track the total being rewarded
-						reward_tracker.total_already_rewarded = reward_tracker
-							.total_already_rewarded
-							.saturating_add(reward_amount_per_oracle);
+				if !reward_amount_per_asset.is_zero() {
+					let rewarded_oracles_total_staked: T::Balance = rewarded_oracles
+						.iter()
+						.map(|account| OracleStake::<T>::get(&account.0).unwrap_or_default())
```

Comment:
why is getting the default reasonable? can we add a comment?

Reply:
to get paid reward, user needs to submit price, to submit price user needs to have a stake.
the user can potentially submit the price and then remove stake in one block, and then rewards are distributed on_initialize of the next block. so in this case we should take 0 as its stake